src/utils/utils.js

```diff
@@ -295,8 +295,25 @@ export const getconstructorParams = (abiConstructor, state, vals, crowdsaleNum)
     return params;
 }
 
-export function roundToDecimals(n, input) {
-  return 1.0 / n * Math.ceil(n * input)
+export const roundToDecimals = (n, input) => Math.floor10(input, n)
+
+const decimalAdjust = (type, value, exp) => {
+    if (typeof exp === 'undefined' || +exp === 0) {
```

Comment:
1) why would exp be equal to the string 'undefined'? This should most likely be the value undefined.
2) what are you trying to do with +exp? if its number coersion, why not make it a number when it is passed in initially?
3) what does having an undefined exp or an exp equal to 0 mean in words? The logic contained within the if conditional should be extracted to a variable as well.

Reply:
Let's use  Math.floor10 and decimalAdjust. It is well-known solution: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round